From 31b934e4de207dc34532c5f0bb14aca41d880c40 Mon Sep 17 00:00:00 2001
From: Pawit Pornkitprasan <p.pawit@gmail.com>
Date: Wed, 25 Dec 2013 07:50:51 +0100
Subject: [PATCH 019/192] Camera2: Preview needs to be stopped when changing
 resolution

This was done in startPreview as a hack, but was removed in
f729b85e12cfa2674048d8cce2f724014b8fc910. Specifically add it
back when the resolution is changed.

Change-Id: I306abd788bafdff4e6fba7eec4e6ff05bb3cc47c
---
 src/com/android/camera/PhotoModule.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 116d7e301..2042f365c 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -2916,6 +2916,12 @@ public class PhotoModule
         mErrorCallback.setActivity(mActivity);
         mCameraDevice.setErrorCallback(mErrorCallback);
 
+        // Preview needs to be stopped when changing resolution
+        if (mRestartPreview && mCameraState != PREVIEW_STOPPED && mCameraState != INIT) {
+            stopPreview();
+            mRestartPreview = false;
+        }
+
         if (!mSnapshotOnIdle) {
             mFocusManager.setAeAwbLock(false); // Unlock AE and AWB.
         }
-- 
2.17.1

